ping/ping.c

```diff
@@ -249,6 +250,30 @@ static int parsetos(char *str)
 	return (tos);
 }
 
+/* Set Identifier */
+static int parse_ident(char* str)
+{
+	const char *cp;
+	unsigned long val;
+	char *ep;
+
+	/* handle both hex and decimal values */
+	if (str[0] == '0' && (str[1] == 'x' || str[1] == 'X')) {
+		cp = str + 2;
+		val = (int)strtoul(cp, &ep, 16);
+	} else
+		val = (int)strtoul(str, &ep, 10);
```

Comment:
I guess handling also hexadecimal is the reason why you haven't used `strtoul_or_err()`. I don't like single option which has numeric parameter behaves differently than the rest (and you haven't mention it in man and help). I wonder if it's safe to add hex handling to `strtoul_or_err()`. But you don't have to bother with it, I can add it myself later.

Also nit: please add brackets also around `else`. `ping` code uses really ancient code style, I'd prefer for new code to use linux kernel code style https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces:

> This does not apply if only one branch of a conditional statement is a single statement; in the latter case use braces in both branches:

```
if (condition) {
        do_this();
        do_that();
} else {
        otherwise();
}
```

Reply:
TBH I just copy-pasted parse_tos function :sweat_smile: 
I can use strtoul_or_err if you like, but I feel that hex input may be useful.

Reply:
Right, could you please use `strtoul_or_err()` now? I'd be for adding hex to `strtoul_or_err()` after the release (trying to minimize changes before release).